### PR TITLE
Updated extract_statement_period function

### PR DESCRIPTION
**Pull request by Harsha2319**
changed the regular expression  date_pattern to solve the date extraction issue.

### CODE REVIEW

File: src/statement_extraction/workflow_001.py [hunk 1]
```diff
@@ -85,12 +85,13 @@ def extract_statement_period(self) -> None:
...
         """
         # Regular expression pattern for dates
-        date_pattern = "[0-1][0-9]/[0-3][0-9]/[1-3][0-9]*3"
+        date_pattern = "\d{2}/\d{2}/\d{4}"
 
         # Iterates across lines in page 3 (transaction page).
         for line in self.document_text["page_2"]:
             # Checks for line containing "open to close date" and gets the statement start & end date from the next line.
             if "open to close date" in line.lower():
+                print(re.findall(date_pattern, line))
```
Comment (preetham-ganesh) [inline src/statement_extraction/workflow_001.py:94]: Remove line.

File: src/statement_extraction/workflow_001.py [hunk 2]
```diff
@@ -330,6 +331,43 @@ def extract_statement_transactions(self) -> None:
             "statement_transactions"
         ] = statement_transactions
 
+    def convert_transaction_date_format(self) -> None:
```
Comment (preetham-ganesh) [inline src/statement_extraction/workflow_001.py:334]: Update function name to 'preprocess_transaction_date'. Remove for loop, and call function during each transaction's extraction.

**Review by preetham-ganesh — CHANGES_REQUESTED** [review]

File: src/statement_extraction/workflow_001.py (hunk 1 above)
Comment (Harsha2319) [reply src/statement_extraction/workflow_001.py]: Removed print statement.

File: src/statement_extraction/workflow_001.py (hunk 2 above)
Comment (Harsha2319) [reply src/statement_extraction/workflow_001.py]: Updated the function & included the functionality as requested.

**Review by preetham-ganesh — APPROVED** [review]
LGTM!

**Pull request merged**